server/cache.go

```diff
@@ -341,6 +341,27 @@ func (c *clusterInfo) updateStoreStatus(id uint64) {
 	c.Stores.SetRegionCount(id, c.Regions.GetStoreRegionCount(id))
 }
 
+func (c *clusterInfo) updateStorePendingPeerCount(origin []*metapb.Peer, new []*metapb.Peer) {
```

Comment:
I think you can use `core.RegionsInfo` to trace pending peers count of each store.